### PR TITLE
feat: image-source room impulse responses and steady-state room field

### PR DESCRIPTION
**Pull request by jmrplens**
## Summary

Room acoustics by image sources and the steady-state field for the room domain, clean-room from Kuttruff (Room Acoustics 6e), Vorländer (Auralization) and Bies, Hansen and Howard.

- `image_source_rir`: the synthetic room impulse response of a rectangular room by the image-source method, enumerating the image lattice to a configurable order with the per-wall reflection count, per-band or broadband wall absorption and optional air attenuation, and assembling g(t) as a sum of delayed, attenuated deltas. Frozen result carrying the exact sub-integer image table (times, distances, orders, amplitudes, positions) with a reflectogram `.plot()`, plus `audible_image_count` and `reflection_density` helpers.
- `room_constant`, `critical_distance`, `schroeder_frequency`, `steady_state_spl` and `steady_state_field`: the classical steady-state relations, with a frozen result splitting the direct, reverberant and total field versus distance and its `.plot()`.

## Validation

Anchored on exact geometry (direct and first-reflection amplitudes and delays to machine precision), the Kuttruff image count (1561 images through order 10), the reflection-density growth law, and the Eyring T60 recovered from the synthetic RIR's reverberant decay (with a documented finding that an elongated room decays slower than Eyring, the Fitzroy/Arau regime). The FDTD 2D solver already in the library is used as an independent numerical cross-check: the rigid-wall echo matches the first image delay, and a uniform-damping T60 comes back at 0.588 s against 0.600 s analytic. The steady-state relations match their closed forms, including the critical distance where the direct and reverberant fields are equal by definition. Seven new conformance checks (319 total, byte-exact report).

## Checks

ruff, mypy, bandit, full pytest (3790 passed, no new warnings), conformance byte-exact, figures within tolerance (848, one new in four variants), api-docs with no drift, curated API table 721 names, site i18n parity and full build clean. Adversarially reviewed; both findings applied, plus proactive guards for coincident source/receiver and the length-6 absorption ambiguity.

<a href="https://cubic.dev/pr/jmrplens/phonometry/pull/229?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Added rectangular-room image-source modeling with reflection analysis, optional air attenuation, broadband/per-band impulse responses, and reflectogram plotting.
  * Added steady-state room-field modeling (room constant, critical distance, Schroeder frequency) with direct/reverberant/total sound levels.
  * Added plotting support for both image-source reflections and steady-state sound fields.
* **Documentation**
  * Added a new English/Spanish guide plus updated README navigation, API reference pages, theory coverage, and expanded references/citations.
* **Validation**
  * Expanded conformance reporting and added comprehensive automated tests for core formulas, reflectograms, and edge cases.

### CODE REVIEW

File: src/phonometry/_plot/room.py [hunk 1]
```diff
@@ -503,3 +506,111 @@ def plot_excitation(
...
+                    cmap="viridis", s=14, zorder=3, label="Reflections")
+    ax.vlines(ms[ref_mask], -120.0, level[ref_mask], color=_C_EDGE, lw=0.4,
+              alpha=0.4, zorder=2)
+    ax.vlines(ms[order0], -120.0, level[order0], color=_C_PRIMARY, lw=1.6,
+              zorder=4)
+    ax.plot(ms[order0], level[order0], "o", color=_C_PRIMARY, ms=7, zorder=5,
+            label="Direct sound", **kwargs)
+
+    cbar = ax.figure.colorbar(sc, ax=ax, pad=0.02)
+    cbar.set_label("Reflection order")
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply src/phonometry/_plot/room.py]: Applied in d3c346e: plot() crashed on a direct-only reflectogram (empty scatter, no colorbar limits); guarded with a check for any reflections plus a regression test.

File: src/phonometry/room/steady_field.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+    """
+    lw = np.asarray(sound_power_level, dtype=np.float64)
+    r = np.asarray(distance, dtype=np.float64)
+    rc = np.asarray(room_constant, dtype=np.float64)
+    directivity = require_positive(directivity, "directivity")
+    if np.any(r <= 0.0) or not np.all(np.isfinite(r)):
+        raise ValueError("'distance' must be positive and finite.")
+    if np.any(rc <= 0.0) or not np.all(np.isfinite(rc)):
+        raise ValueError("'room_constant' must be positive and finite.")
+    bracket = directivity / (4.0 * np.pi * r**2) + 4.0 / rc
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply src/phonometry/room/steady_field.py]: Rename applied in d3c346e (rc was the critical distance elsewhere in the module, confusing); the finiteness checks you suggested were already present, so that part is redundant.

File: src/phonometry/room/image_source.py [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+
+    :param time: Time after the direct sound, s (scalar or array).
+    :param volume: Room volume ``V``, m3.
+    :param speed_of_sound: Speed of sound ``c``, m/s.
+    :return: ``dN/dt`` in reflections per second.
+    """
+    volume = require_positive(volume, "volume")
+    speed_of_sound = require_positive(speed_of_sound, "speed_of_sound")
+    t = np.asarray(time, dtype=np.float64)
+    out = 4.0 * np.pi * speed_of_sound**3 * t**2 / volume
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply src/phonometry/room/image_source.py]: Applied in d3c346e: reflection_density clamps t to zero before squaring, so negative times no longer produce a spurious positive density.

File: tests/room/test_image_source.py [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+    freqs = [250.0, 500.0, 1000.0]
+    alpha = np.array([[0.1] * 6, [0.2] * 6, [0.4] * 6]).T  # (6, 3)
+    res = image_source_rir((5.0, 4.0, 3.0), (1.2, 1.1, 1.3), (3.5, 2.6, 1.7),
+                           alpha, fs=16000, max_order=6, frequencies=freqs)
+    assert res.ir.ndim == 2 and res.ir.shape[0] == 3
+    assert res.amplitudes.shape[0] == 3
+    assert res.frequencies is not None and res.frequencies.size == 3
+    # A more absorbing band decays to a lower total energy.
+    energies = np.sum(res.ir**2, axis=1)
+    assert energies[0] > energies[2]
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply tests/room/test_image_source.py]: Applied in d3c346e: the assertion now checks the full monotonic chain across the three bands.

File: tests/room/test_image_source.py [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+def test_length_six_is_per_band_with_six_frequencies() -> None:
+    # With six declared bands, a length-6 vector is a per-band curve.
+    freqs = [125.0, 250.0, 500.0, 1000.0, 2000.0, 4000.0]
+    res = image_source_rir((5.0, 4.0, 3.0), (1.2, 1.1, 1.3), (3.5, 2.6, 1.7),
+                           np.array([0.1, 0.15, 0.2, 0.3, 0.45, 0.6]),
+                           fs=16000, max_order=5, frequencies=freqs)
+    assert res.ir.shape[0] == 6
+    # More absorption at high frequency -> less total energy there.
+    energies = np.sum(res.ir**2, axis=1)
+    assert energies[0] > energies[-1]
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply tests/room/test_image_source.py]: Applied in d3c346e: the assertion now checks strict monotonic decrease across all six bands.

File: docs/references.md [hunk 6]
```diff
@@ -357,8 +371,10 @@ it; the list grows as guides gain their References sections.
   control* (5th ed.). CRC Press. ISBN 978-1-4987-2405-0.
   [doi:10.1201/9781351228152](https://doi.org/10.1201/9781351228152).
   The engineering-noise-control reference; its Appendix D collects the
-  porous-material coefficient sets and layered-construction recursions.
-  Cited by [Porous and Multilayer Absorbers](porous-absorbers.md).
+  porous-material coefficient sets and layered-construction recursions, and
+  its Chapter 6 the steady-state room field and room constant.
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply docs/references.md]: Applied in d3c346e: the sentence now completes with 'and its Chapter 6 covers the steady-state room field and the room constant.'

File: docs/room-image-sources.md [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+room impulse response is the sum of the direct sound and one delayed,
+attenuated impulse per image (Kuttruff Equations (4.4)–(4.5),
+`g(t) = Σ_n A_n δ(t − t_n)`).
+
+Image `i` at distance `r_i` from the receiver arrives at `t_i = r_i / c`
+(Vorländer Equation (11.38)) with amplitude
+
+```
+A_i = [ Π_walls R_wall^(reflections there) ] · exp(−m r_i / 2) / (4 π r_i)
+```
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply docs/room-image-sources.md]: Applied in d3c346e: the five formula fences are tagged text (MD040); the python and closing fences are unchanged.

File: docs/theory-rooms-buildings.md [hunk 8]
```diff
@@ -56,6 +56,30 @@ $$
...
+
+The initial slope of the specular reverberant energy reproduces the **Eyring** rate $T = -24 V \ln 10 / (c S \ln(1 - \bar\alpha))$ (Kuttruff Eq. 5.23), because the mean reflection rate $cS/4V$ equals $\tfrac{c}{2}(1/L_x + 1/L_y + 1/L_z)$. The match is exact only near cubic geometry; an elongated room sustains energy along its long axis, so the pure specular decay runs slower than Eyring's diffuse-field estimate (the anisotropy the Fitzroy/Arau-Puchades models correct). The model is specular only — no diffraction or diffuse scattering — and exact only for real, angle-independent reflection factors.
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply docs/theory-rooms-buildings.md]: Applied in d3c346e: T is the Eyring reverberation time, not a rate; reworded to 'the initial decay rate recovers the Eyring reverberation time T' across the affected docs.

File: docs/theory-rooms-buildings.md [hunk 9]
```diff
@@ -56,6 +56,30 @@ $$
...
+
+### Steady-state room field (Bies 6.4, Kuttruff 5.6)
+
+A source of constant power sets up a steady level made of a direct field falling with distance and a diffuse **reverberant** field that is (approximately) uniform. With the **room constant** $R = S\bar\alpha/(1-\bar\alpha)$ (Bies Eq. 6.44) and the directivity factor $Q$,
+
+$$
+L_p = L_W + 10 \lg\!\left( \frac{Q}{4\pi r^2} + \frac{4}{R} \right) \left[ + 10 \lg\frac{\rho c}{400} \right] \quad \text{(Bies Eq. 6.43)},
+$$
+
+the optional last term (about $+0.14$ dB at 20 °C) correcting a characteristic impedance away from 400 Pa·s/m. The **critical distance** $r_c = \sqrt{Q R / 16\pi}$ is where the two fields cross; Kuttruff's reverberation distance (Eq. 5.44, $r_c = \sqrt{A/16\pi}$ for $Q=1$) uses the Sabine area $A = S\bar\alpha$ instead of $R = A/(1-\bar\alpha)$, the two coinciding for small $\bar\alpha$. The **Schroeder frequency** $f_s = 2000\sqrt{T/V}$ (Kuttruff Eq. 3.44) marks the boundary below which discrete room modes rule and these diffuse-field relations fail. See the [Image sources and steady-state field guide](room-image-sources.md).
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply docs/theory-rooms-buildings.md]: Applied in d3c346e: the Schroeder frequency is now described as a heuristic modal-to-diffuse crossover, not a hard cutoff, with a note to check band by band (EN, ES, theory and the GitHub doc).

File: llms-full.txt [hunk 10]
```diff
@@ -4803,6 +4804,224 @@ decays and the standards' own parameter definitions in the
...
+    max_order=12,
+)
+
+print(res.ir.shape)                          # (n_samples,) broadband RIR
+print(round(res.direct_time * 1000, 2))      # direct-sound arrival, ms
+print(res.times.size, room.audible_image_count(12) + 1)  # images + direct source
+
+# Feed the synthetic RIR straight into the ISO 3382 decay analysis.
+params = room.room_parameters(res.ir, res.fs, limits=None)
+print(round(float(params.t30[0]), 2))        # reverberation time, s
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply llms-full.txt:4886]: Premise did not reproduce, but the spirit is applied in d3c346e. Verified empirically: with max_order=12 the example yields t30_valid=True and T30=0.28 s (not NaN/invalid). That 0.28 s is order-truncated (Eyring is about 0.93 s), so the source now prints t30_valid and notes that T30 grows toward Eyring with max_order; llms regenerated.

File: docs/references.md [hunk 11]
```diff
@@ -357,8 +371,10 @@ it; the list grows as guides gain their References sections.
   control* (5th ed.). CRC Press. ISBN 978-1-4987-2405-0.
   [doi:10.1201/9781351228152](https://doi.org/10.1201/9781351228152).
   The engineering-noise-control reference; its Appendix D collects the
-  porous-material coefficient sets and layered-construction recursions.
-  Cited by [Porous and Multilayer Absorbers](porous-absorbers.md).
+  porous-material coefficient sets and layered-construction recursions, and
+  its Chapter 6 covers the steady-state room field and the room constant.
+  Cited by [Porous and Multilayer Absorbers](porous-absorbers.md) and
+  [Image sources and the steady-state room field](room-image-sources.md).
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply docs/references.md:377]: Applied in 3590d1a: the Bies 5th-edition year is now 2017 (the DOI and publisher date) in docs/references.md. Fixed the same stale 2018 in the three other places it appeared for consistency (docs/porous-absorbers.md and both porous-absorbers site pages); the room, ground-barriers and theory entries were already 2017.
